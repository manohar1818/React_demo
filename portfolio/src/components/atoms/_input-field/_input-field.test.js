import React from 'react';
import { create } from "react-test-renderer";
import { render ,fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import InputField from './_input-field';

describe("InputField component",() =>{
    test("Matches the snapshot",() => {
        const inputField = create(<InputField />);
        expect(inputField.toJSON()).toMatchSnapshot();

    })
    test("inputField element is rendered",() => {
        const { getByTestId, getByText } = render(<InputField />);
        expect(getByTestId("inputField")).toBeInTheDocument();
        
    });
    test("on change of input",() => {
        const mockOnChange = jest.fn();
        const { getByTestId} = render(<InputField value="name"  onChange={mockOnChange}/>);        
        fireEvent.change(getByTestId("name"),{ target: { value: 'ram' } });
        
        expect(mockOnChange).toHaveBeenCalledTimes(1);
    });
   

});