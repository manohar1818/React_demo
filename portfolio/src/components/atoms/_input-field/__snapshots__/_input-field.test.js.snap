// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component Matches the snapshot 1`] = `
<div
  className="makeStyles-styleInputFields-1"
  data-testid="inputField"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline"
    data-testid="inputAtom"
    onClick={[Function]}
    required={true}
    style={
      Object {
        "color": "#000",
      }
    }
  >
    <input
      className="MuiInputBase-input MuiInput-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="text"
    />
  </div>
</div>
`;
